### PR TITLE
feat(config): add config instead of secrets

### PR DESCRIPTION
**Pull request by Oupsla**
Hello 👋 

The goal of this PR is to use config files instead of storing everything in .env files (this allows to made complex structure instead of "just" strings in .env files, and to use .env files for secrets only).

For info, I have gone a bit further in my own work (but didn't have pushed this in this PR), by requiring a config file based on the NODE_ENV variable. This allows to have one config file for each environnement (and each environnement config file has to declare all variables, that can't be the case with .env files). Tell me if something like is interresting and I can open another PR 😄 

Have a nice weekend !

### CODE REVIEW

**Comment by marcjulian** [comment]
@Oupsla thanks for your contribution this looks way cleaner than using the `.env` variables everywhere! I will review your PR asap!! Makes also more sense to *just* use `.env` for secrets rather than all configurations!

**Comment by marcjulian** [comment]
@Oupsla wish you a great weekend too!

File: src/configs/config.interface.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+export interface GraphqlConfig {
+  playgroundEnabled: boolean;
+  debug: boolean;
+  schemaDestination: string;
+}
+
+export interface SecurityConfig {
+  expiresIn: string;
+  refreshIn: string;
+  bcryptSaltOrRound: number;
```
Comment (marcjulian) [inline src/configs/config.interface.ts:34]:
```suggestion
  bcryptSaltOrRound: string | number;
```

File: src/configs/config.interface.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  description: string;
+  version: string;
+  path: string;
+}
+
+export interface GraphqlConfig {
+  playgroundEnabled: boolean;
+  debug: boolean;
+  schemaDestination: string;
+}
```
Comment (marcjulian) [inline src/configs/config.interface.ts:29]:
```suggestion
  schemaDestination: string;
  sortSchema: boolean;
}
```

File: src/configs/config.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    title: 'Nestjs FTW',
+    description: 'The nestjs API description',
+    version: '1.5',
+    path: 'api',
+  },
+  graphql: {
+    playgroundEnabled: true,
+    debug: true,
+    schemaDestination: './src/schema.graphql',
+  },
```
Comment (marcjulian) [inline src/configs/config.ts:21]:
```suggestion
    schemaDestination: './src/schema.graphql',
    sortSchema: true,
  },
```

File: src/app.module.ts [hunk 4]
```diff
@@ -8,23 +8,27 @@ import { PostModule } from './resolvers/post/post.module';
...
-          configService.get('PLAYGROUND_ENABLE') === '1' ? true : false,
-        context: ({ req }) => ({ req }),
-      }),
+      useFactory: async (configService: ConfigService) => {
+        const graphqlConfig = configService.get<GraphqlConfig>('graphql');
+        return {
+          buildSchemaOptions: {
+            numberScalarMode: 'integer',
+          },
+          sortSchema: true,
```
Comment (marcjulian) [inline src/app.module.ts:24]:
```suggestion
          sortSchema: graphqlConfig.sortSchema,
```

**Review by marcjulian — CHANGES_REQUESTED** [review]
LGTM 👍  thanks for simplifying configs!

I requested only small changes.

File: src/configs/config.interface.ts (hunk 1 above)
Comment (marcjulian) [reply src/configs/config.interface.ts:34]: Should be possible to provide a salt as `string`

**Comment by marcjulian** [comment]
@Oupsla thanks again for your PR 👍 do you like to create another PR one for requiring a configuration file for each environment?

**Comment by Oupsla** [comment]
@marcjulian, while working on a project (started with this template), I have noticed that Prisma has also a .env file, I have figured out a solution to use .env files at the root folder (using dotenv-cli). I'll make another PR with that, if you think that could be interesting.



> @Oupsla thanks again for your PR 👍 do you like to create another PR one for requiring a configuration file for each environment?

No problem, glad to help ! 😄  Yes I'll create a PR for that

**Comment by marcjulian** [comment]
Awesome thank you, that will definitely be helpful for others as well.

**Comment by Oupsla** [comment]
(Also if you don't mind adding `hacktoberfest` in topics of this repo, so my PR's could count, cause it is hard to fill the counter after this update : https://hacktoberfest.digitalocean.com/hacktoberfest-update 😄 
Sorry for the noise ! )

**Pull request merged**